client/blocks/post-share/index.jsx

```diff
@@ -277,7 +278,7 @@ class PostShare extends Component {
 				id: ID,
 				type: 'published-action',
 				title: message,
-				date,
+				date: moment( date ),
```

Comment:
In the objects passed as items in `events`, the `date` field is expected to be `moment` or `Date` instance or a number timestamp. See here: https://github.com/Automattic/wp-calypso/blob/master/client/components/date-picker/index.jsx#L182-L192

That means we don't need `moment` here and `new Date( date )` is just fine. What is the `date` destructured from input in this `map` statement? A string? We could also patch `DatePicker.getDateInstance` to accept a string, too.

Reply:
I've been unsuccessfully trying to test this for about half an hour no, and am no closer to being able to tell you what `date` is here. This is a good example of where types would be useful 😞

Is there a good way of testing this without creating and linking a social media account?

Reply:
From what I can tell, the original code was wrong, and this shouldn't be `date`, but rather `share_date` or `shareDate`, which according to the JSON schema is an integer (so UNIX timestamp, perhaps?).

Reply:
@jsnajdr: Let me know if you find anything out here, happy to make changes. Otherwise, this is one place I'd suggest keeping the current high-level approach of trying to preserve the current behaviour, rather than going too in-depth.

Reply:
@sgomes Would you mind if I push a few commits if I find anything during testing? I have a few Facebook testing pages connected to my account.

Reply:
@jsnajdr: No, please go right ahead! And thank you! 🙇 

I'll work on a separate PR in the meantime, tracking down the a few more bits of `i18n-calypso` `moment` in state that I found.